developer-docs/latest/concepts/sso.md

```diff
@@ -0,0 +1,320 @@
+          }),
+        },
+      ],
+    },
+  },
+});
+```
+:::
+
+::: tab Discord
+
+Using: [passport-discord](https://github.com/nicholastay/passport-discord#readme)
+
+```bash
+yarn add passport-discord
+```
+
+`/config/server.js`
+
+```jsx
+"use strict";
+
+const DiscordStrategy = require("passport-discord");
+
+module.exports = ({ env }) => ({
+  // ...
+  admin: {
+    // ...
+    auth: {
+      // ...
+      providers: [
+        {
+          uid: 'discord',
+          displayName: 'Discord',
+          icon: 'https://cdn0.iconfinder.com/data/icons/free-social-media-set/24/discord-512.png',
+          createStrategy: strapi =>
+            new DiscordStrategy(
+              {
+                clientID: env('DISCORD_CLIENT_ID'),
+                clientSecret: env('DISCORD_SECRET'),
+                callbackURL: strapi.admin.services.passport.getStrategyCallbackURL(
+                  'discord'
+                ),
+                scope: ['identify', 'email'],
+              },
+              (accessToken, refreshToken, profile, done) => {
+                done(null, {
+                  email: profile.email,
+                  username: `${profile.username}#${profile.discriminator}`,
+                });
+              }
+            ),
+        },
+      ],
+    },
+  },
+});
+```
+:::
+::::
+
+## Advanced Customization
+
+### Admin Panel URL
+
+If your administration panel lives on a different host/port than your Strapi server, you will need to modify the admin URL.
+To do so, head to your `/config/server.js` configuration file and tweak the `admin.url` field.
+
+For example, if your admin application has been started on `localhost:4000`, your configuration will look like the following:
+
+```javascript
+{
+  ...
+  admin: {
+    url: 'http://localhost:4000/admin',
+    ...
+  }
+}
+```
+
+### Custom Logic
+
+In some scenarios, you will want to write additional logic for your connection workflow such as:
+- Restricting connection and registration for a specific domain
+- Triggering actions on connection attempt
+- Analytics
+
+The easiest way to do so is to plug into the verify function of your strategy and write some code.
+
+For example, if you want to allow only people with an official strapi.io email address, you can instantiate your strategy like this:
+
+```javascript
+const strategyInstance = new Strategy(
+  configuration,
+  (email, done) => {
+    if (!email.endsWith('@strapi.io')) {
+      done(new Error('Forbidden email address'));
+      return;
+    }
+    
+    done(null, { email });
+  },
+);
+```
+
+### Authentication Events
+
+The SSO feature adds a new [authentication event](configurations.html#available-options): `onSSOAutoRegistration`.
+
+This event is triggered whenever a user is created using the auto-register feature added by SSO.
+It contains the created user (`event.user`), and the provider used to make the registration (`event.provider`).
+
+Example:
+
+```javascript
+module.exports = () => ({
```

Comment:
While we do understand this is part of the server.js, we should make that clear with a file path and the typical snippet comments `// ...` to make it very clear where it's to be used.

The best method for clarifying documentation is to consider someone with absolutely no previous knowledge of the system should be able to reasonably determine  the location and requirements beforehand and typically at a first glance.

Reply:
You're right, I've modified the code examples to reflect your suggestion.